File: ios/Classes/PlatformEnums.swift
import Foundation

/// 平台事件枚举
enum PlatformEvent: String {
    case loadCharacterState = "loadCharacterState"
    case didUpdatedConnectionState = "didUpdatedConnectionState"
    case didUpdatedConversationState = "didUpdatedConversationState"
    case didUpdatedPlayerState = "didUpdatedPlayerState"
    case playerDidEncounteredError = "playerDidEncounteredError"
}

enum PlatformSetUpState: String {
    case notSetUp = "notSetUp"
    case successed = "successed"
}

/// 平台方法枚举
enum PlatformMethod: String {
    case loadCharacterData = "loadCharacterData"
    case preloadCharacterData = "preloadCharacterData"
    case start = "start"
    case close = "close"
    case interrupt = "interrupt"
    case sendAudioData = "sendAudioData"
    case setVolume = "setVolume"
    case deleteCharacterAssets = "deleteCharacterAssets"
    case deleteAllCharacterAssets = "deleteAllCharacterAssets"
}

File: android/src/main/kotlin/ai/spatialwalk/sp_character_plugin/PlatformEnums.kt
package ai.spatialwalk.sp_character_plugin

/**
 * Platform event enums
 */
enum class PlatformEvent(val value: String) {
    LOAD_CHARACTER_STATE("loadCharacterState"),
    DID_UPDATED_CONNECTION_STATE("didUpdatedConnectionState"),
    DID_UPDATED_CONVERSATION_STATE("didUpdatedConversationState"),
    DID_UPDATED_PLAYER_STATE("didUpdatedPlayerState"),
    PLAYER_DID_ENCOUNTERED_ERROR("playerDidEncounteredError")
}

/**
 * Platform setup state
 */
enum class PlatformSetUpState(val value: String) {
    NOT_SET_UP("notSetUp"),
    SUCCEEDED("successed")
}

/**
 * Platform methods
 */
enum class PlatformMethod(val value: String) {
    LOAD_CHARACTER_DATA("loadCharacterData"),
    PRELOAD_CHARACTER_DATA("preloadCharacterData"),
    START("start"),
    CLOSE("close"),
    INTERRUPT("interrupt"),
    SEND_AUDIO_DATA("sendAudioData"),
    SET_VOLUME("setVolume"),
    DELETE_CHARACTER_ASSETS("deleteCharacterAssets"),
    DELETE_ALL_CHARACTER_ASSETS("deleteAllCharacterAssets");

    companion object {
        fun fromValue(value: String): PlatformMethod? {
            return entries.find { it.value == value }
        }
    }
}

